v2.go

```diff
@@ -83,63 +88,67 @@ func parseVersion2(reader *bufio.Reader) (header *Header, err error) {
 	}
 	header.TransportProtocol = AddressFamilyAndProtocol(b14)
 	//  UNSPEC is only supported when LOCAL is set.
-	if header.TransportProtocol == UNSPEC {
-		if header.Command != LOCAL {
-			return nil, ErrUnsupportedAddressFamilyAndProtocol
-		}
-		// Ignore everything else.
-		return header, nil
+	if header.TransportProtocol.IsUnspec() && (header.Command != LOCAL || header.TransportProtocol != UNSPEC) {
+		return nil, ErrUnsupportedAddressFamilyAndProtocol
 	}
 
 	// Make sure there are bytes available as specified in length
 	var length uint16
 	if err := binary.Read(io.LimitReader(reader, 2), binary.BigEndian, &length); err != nil {
 		return nil, ErrCantReadLength
 	}
-	if !header.validateLength(length) {
+
+	// There's no need to validate for UNSPEC since length always >= 0
+	if header.TransportProtocol != UNSPEC && !header.validateLength(length) {
 		return nil, ErrInvalidLength
 	}
 
+	if length == 0 {
```

Comment:
Please, add a comment stating that if there are no TLVs present, length is zero and we can return early, otherwise, the code continues so TLVs reading is attempted.